### PR TITLE
Clarify word separator docs about shortcuts

### PR DESCRIPTION
**Pull request by kitbdev**
Default and custom word separators affect all word-related actions, not just `ctrl+left/right`. These can be rebound, so I mention the action instead of the shortcut. I don't mention all of them because it would get too long (backspace word, caret word right, and select word under caret).
Also tweaked the code_complete_enabled description to use the action as well, which can be used regardless of the setting.

### CODE REVIEW

**Comment by Mickeon** [comment]
Conflicted on directly using the action names.
On most users' mind, the shortcut comes first, and the associated action comes second. It does not work like this, sure, but it's much easier to imagine and even play around with in your text editor of choice.

**Comment by kitbdev** [comment]
How about something like:

If [code]false[/code], using word selections or operations such as double-clicking or holding [kbd]Ctrl[/kbd] ([kbd]Cmd[/kbd] on macOS) while pressing [kbd]left[/kbd], [kbd]right[/kbd], [kbd]backspace[/kbd], or [kbd]delete[/kbd], will ...

**Comment by Mickeon** [comment]
I like that more, although it's quite the mouthful. I would recommend rearranging the sentences into something more akin to:
```
If [code]false[/code], when selecting a word the caret stops moving only if a space or punctuation is detected. Word selection operations include double-clicking or holding [kbd]Ctrl[/kbd] ([kbd]Cmd[/kbd] on macOS) while pressing [kbd]left[/kbd], [kbd]right[/kbd], [kbd]backspace[/kbd], or [kbd]delete[/kbd].
```
It could also be nice to reword this to explain what [code]true[/code] does first, especially if that's the default now.

Note that I'm not always against referring to the actions. I just don't believe it's nice to in the Editor Settings, given there's no direct way to point users to them in the Input Map.

**Comment by kitbdev** [comment]
Reworded to show true value first, and hopefully its more clear.
Changed example in custom_word_separators, since # and ! are always treated as separated so there is no point to having them be custom separators.
Made sure its clear that the default and custom settings work together additively.
For use_default_word_separators, I tried to find what characters are used when it is false, but I can't easily just put all of them in the description, it contains some characters from the punctuation tables, but not all of them. Instead I marked some notable characters that are not included.

File: doc/classes/EditorSettings.xml [hunk 1]
```diff
@@ -1091,7 +1091,7 @@
 			The delay in seconds after which autocompletion suggestions should be displayed when the user stops typing.
 		</member>
 		<member name="text_editor/completion/code_complete_enabled" type="bool" setter="" getter="">
-			If [code]true[/code], code completion will be triggered automatically after [member text_editor/completion/code_complete_delay]. If [code]false[/code], you can still trigger completion manually by pressing [kbd]Ctrl + Space[/kbd] ([kbd]Cmd + Space[/kbd] on macOS).
+			If [code]true[/code], code completion will be triggered automatically after [member text_editor/completion/code_complete_delay]. Even if [code]false[/code], code completion can be triggered manually with the [code]ui_text_completion_query[/code] action (default [kbd]Ctrl + Space[/kbd] or [kbd]Cmd + Space[/kbd] on macOS).
```
Comment (Mickeon) [inline doc/classes/EditorSettings.xml:1094]:
```suggestion
			If [code]true[/code], code completion will be triggered automatically after [member text_editor/completion/code_complete_delay]. Even if [code]false[/code], code completion can be triggered manually with the [code]ui_text_completion_query[/code] action (by default [kbd]Ctrl + Space[/kbd] or [kbd]Cmd + Space[/kbd] on macOS).
```

**Review by Mickeon — APPROVED** [review]
Pretty good now, assuming it's accurate.

**Comment by akien-mga** [comment]
Thanks!
